src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +33,32 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${todo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => setSelectedTodoId(0)}
```

Comment:
You are generating keys on render by using setSelectedTodoId(0). This is not recommended because it can lead to performance issues. Instead, you should generate keys when the data is initially created. This is a performance issue.